src/analyses/base.ml

```diff
@@ -1865,13 +2067,14 @@ struct
         | _ -> false
       in
       let itv = (* int abstraction for tv *)
-        let ik = Cilfacade.get_ikind_exp exp in
+        (* when using floats without explicit cast, we can actually get a non-integer type *)
+        let ik = try Cilfacade.get_ikind_exp exp with Invalid_argument _ -> IInt in
```

Comment:
Is it correct to use `IInt` here in case the expression is not an integer?

Reply:
Yes it appears harmless here, the only cases where we go here is for pointer sand floats, and they are then correctly handled late in the invariant by` FD.of_int` and `AD.of_int`.

Reply:
Actually after discussing this further with @jerhard, it does in fact turn out to be problematic: In the true case, we erroneously claim that the float must be in the range of `IInt` (and neither `NaN` nor `+/- Inf`).